File: ET3ALEM_SERVER/BusinessEntities/Models/Question/OrderQuestion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using BusinessEntities.Enumerators;
using BusinessEntities.Models.Interfaces;
using Helpers.Extensions;

namespace BusinessEntities.Models
{
    public class OrderQuestion : Question, IAfterSaveAction, IParentEntity
    {
        public OrderQuestion()
        {
            QuestionType = QuestionType.Order;
        }

        public virtual List<OrderedElement> OrderedElements { get; set; }
        public string CorrectOrderIds { get; set; }

        public void PreformAfterSaveAction()
        {
            CorrectOrderIds = string.Join(",", OrderedElements.Select(element => element.Id));
        }

        public IEnumerable<IEnumerable<object>> GetChildEntitiesToDelete()
        {
            return new List<IEnumerable<object>>
            {
                OrderedElements.GetDeletedElements()
            };
        }

        public IEnumerable<IEnumerable<object>> GetChildEntitiesToAdd()
        {
            return new List<IEnumerable<object>> {OrderedElements.GetAddedElements()};
        }

        public IEnumerable<IEnumerable<object>> GetChildEntitiesToUpdate()
        {
            return new List<IEnumerable<object>> {OrderedElements.GetUpdatedElements()};
        }

        public IEnumerable<IEnumerable<object>> GetAllChildEntities()
        {
            return new List<IEnumerable<object>> {OrderedElements};
        }

        public void SetChildEntitiesToNull()
        {
            this.OrderedElements = null;
        }

        public IEnumerable<IEnumerable<object>> RemoveDeletedEntitiesFromChildren()
        {
            var itemsToDelete = OrderedElements.GetDeletedElements().ToList();
            OrderedElements.RemoveAll(choice => choice.Id < 0);
            foreach (var orderedElement in itemsToDelete)
            {
                orderedElement.Id = Math.Abs(orderedElement.Id);
            }
            return new List<IEnumerable<object>> {itemsToDelete};
        }
    }
}

File: ET3ALEM_CLIENT_WEB/src/app/question/Models/order-question.ts
import {Question} from "./question";
import {OrderedElement} from "./ordered-element";
import {Type} from "class-transformer";
import {QuestionType} from "./question-type.enum";

export class OrderQuestion extends Question
{
  @Type(() => OrderedElement)
  OrderedElements: Array<OrderedElement>;
  CorrectOrderIds: String;

  constructor(Id: number = 0, Body: string = '', OrderedElements = [new OrderedElement(), new OrderedElement()], CorrectOrderIds: String = null, Comment: string = null) {
    super(Id, Body, Comment);
    this.QuestionType = QuestionType.OrderQuestion;
    this.OrderedElements = OrderedElements;
    this.CorrectOrderIds = CorrectOrderIds;
  }
}
